### DISCUSSION TITLE
3D Scene - adjust lighting parameters

### DISCUSSION THREAD

**Discussion by andreear-23** [zauberzeug/nicegui#3917] [Ideas / Feature Requests]
### Question

Hi everyone,

Is it possible to adjust the renderer parameters for the `ui.scene` element? I am interesting in changing things like lighting and renderer for my scene. I am loading a GLB mesh with texture but it looks really dark in the scene. 
For example, this is how a GLB looks like loaded with `ui.scene`
![image](https://github.com/user-attachments/assets/74d805ef-3d77-4f80-8c38-d1eae256de12)
and this is how it looks when I load it in another web based gLTF Viewer (which is also using threejs as a backbone)
![image](https://github.com/user-attachments/assets/9e4132a3-f25e-4e19-b8fb-cf37075ef7c4)

In these screenshots the difference is not as bad, but for the GLB models I am using it's really obvious.

Thanks for your help!

**Comment by falkoschindler** [comment]
Hi @andreear-23,

Currently the light is hard-coded:
https://github.com/zauberzeug/nicegui/blob/3b9293db147ba957b074e6d8e879490ca41d2afc/nicegui/elements/scene.js#L101-L104

You can add spot lights like this: 
```py
with ui.scene() as scene:
    scene.sphere()
    scene.spot_light().move(0, 0, 3).rotate(0, math.pi / 2, 0)
```

Removing the two existing lights is a bit tricky, but possible:
```py
with ui.scene() as scene:
    scene.sphere()
    scene.on('init', lambda: ui.run_javascript(
        f'getElement("{scene.id}").scene.children.slice(0, 2).forEach(child => child.removeFromParent())'
    ))
    scene.spot_light().move(0, 0, 3).rotate(0, math.pi / 2, 0)
```
<img width="398" alt="Screenshot 2024-10-24 at 19 25 08" src="https://github.com/user-attachments/assets/eec8751e-258e-4abb-af28-bde2fc7be796">

Let's turn this question into a feature request to improve the API for configuring existing lighting and adding more light objects like ambient or directional light.

**Reply by andreear-23** [reply]
Hi @falkoschindler, thanks a lot for your help.
I guess it is not possible to adjust the hardcoded ambient light either - I tried really quickly to change it in the `scene.js` but it doesn't seem to be as simple as that.
What I am interested in is to display some meshes that have textures with baked-in lighting, so I am trying to reproduce the view I get in Meshlab with the shading turned off. Honestly I don't really know how they configure their renderer.

It would be quite helpful though to have more API functionality for configuring scene lighting. 

Thanks,
Andreea
